packages/typescript/index.js

```diff
@@ -29,6 +29,22 @@ function typescript (options) {
         }
       ]
     },
-    plugins: options ? [new CheckerPlugin(), new TsConfigPathsPlugin({ tsconfig: options.tsconfig, compiler: options.compiler })] : [new CheckerPlugin()]
+    plugins: [
+      new CheckerPlugin()
+    ].concat(
+      options ? [
+        new TsConfigPathsPlugin({ tsconfig: options.tsconfig, compiler: options.compiler })
+      ] : []
+    )
   })
+  
+  return Object.assign(setter, { pre })
+}
+
+function pre (context) {
+  if ('application/x-typescript' in context.fileType.all()) {
```

Comment:
True, but the `if (!(quite~long~thing~here()))` looks shitty as well... How about:

```js
const registeredTypes = context.fileType.all()
if (!('application/x-typescript' in registeredTypes)) {
  ...
}
```